src/main/java/hudson/plugins/jira/JiraBuildAction.java

```diff
@@ -1,73 +1,86 @@
 package hudson.plugins.jira;
 
-import java.net.URL;
-import java.util.HashSet;
-import java.util.Set;
-
 import hudson.model.Action;
 import hudson.model.Run;
 import hudson.plugins.jira.model.JiraIssue;
+import java.io.Serializable;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import jenkins.model.RunAction2;
 import org.kohsuke.stapler.export.Exported;
 import org.kohsuke.stapler.export.ExportedBean;
 
-import javax.annotation.Nonnull;
-
 /**
  * Jira issues related to the build.
  *
  * @author Kohsuke Kawaguchi
  */
 @ExportedBean
-public class JiraBuildAction implements Action {
+public class JiraBuildAction implements RunAction2 {
 
-    public final Run<?, ?> owner;
+  private final HashSet<JiraIssue> issues;
+  private transient Run<?, ?> owner;
```

Comment:
@jglick thanks a lot for helping with this PR!
One question, as your comment about `transient` got me thinking.
Sure, we don't want `issues` to be transient in order for them to be serialized into XML, but isn't it the same with `owner`? I mean, where does `RunAction2` know from to which build it should link it to? 
Is it Jenkins that is saving it somewhere internally?

Reply:
Yes `Run` knows to inject itself into a `RunAction2` both in `addAction` and upon deserialization. The same applies to various other APIs with “owner” references, like `JobProperty`.